mtags/src/main/scala-2/scala/meta/internal/pc/PcInlineValueProviderImpl.scala

```diff
@@ -0,0 +1,189 @@
+package scala.meta.internal.pc
+
+import scala.annotation.tailrec
+
+import scala.meta.internal.mtags.MtagsEnrichments._
+import scala.meta.internal.pc.Definition
+import scala.meta.internal.pc.InlineValueProvider.Errors
+import scala.meta.internal.pc.Reference
+import scala.meta.pc.OffsetParams
+
+import org.eclipse.{lsp4j => l}
+
+final class PcInlineValueProviderImpl(
+    val compiler: MetalsGlobal,
+    val params: OffsetParams
+) extends InlineValueProvider {
+  self =>
+
+  import compiler._
+
+  val pcCollector: PcCollector[(Tree, Option[Tree], Position)] =
+    new PcCollector[(Tree, Option[Tree], Position)](compiler, params) {
+      override def collect(
+          parent: Option[compiler.Tree]
+      )(tree: compiler.Tree, pos: Position): (Tree, Option[Tree], Position) = {
+        val (adjustedPos, _) = adjust(pos)
+        (tree.asInstanceOf[Tree], parent.map(_.asInstanceOf[Tree]), adjustedPos)
+      }
+    }
+
+  val position: l.Position = pcCollector.pos.toLsp.getStart()
+
+  override val text: Array[Char] = pcCollector.text
+
+  def localRefs(
+      tree: Tree
+  ): List[Tree] = {
+    @tailrec
+    def collectNames(
+        symbols: List[Tree],
+        tree: Tree,
+        toTraverse: List[Tree]
+    ): List[Tree] = {
+      val nextToTraverse = toTraverse ++ tree.children
+      val nextSymbols =
+        tree match {
+          case id @ Ident(_) => id :: symbols
+          case s @ Select(_, _) => s :: symbols
+          case _ => symbols
+        }
+      nextToTraverse match {
+        case head :: tail => collectNames(nextSymbols, head, tail)
+        case Nil => nextSymbols
+      }
+    }
+    tree.children match {
+      case head :: tail => collectNames(List(), head, tail)
+      case Nil => Nil
+    }
+  }
+
+  def makeRefs(
```

Comment:
Is this needed for the reference edits? We could potentially use `ShortenedNames.synthesize`, which should be able to figure out how to print a symbol properly and all the edits that are needed. There is a different thing for Scala 3, AutoImportEdits should help there. Those are classes that operate on context based around position, so they will have all the info about conflicts etc.

So essentially, they could give you the ability to see if there is a conflict for any symbol in the tree and it's a well tested logic. You wouldn't need to collect names etc. since that should already be present there.

Reply:
I ended up making my version based on `ShortenedNames.synthesize` and `AutoImportEdits`. I feel like inline already makes a lot of changes, so I didn't want it to add `import` statements as well. Also both of those can only deal well with static elements and can't produce an element such as `this.e`, which is sometimes the desired way of accessing a variable.